app/controllers/users_controller.rb

```diff
@@ -0,0 +1,44 @@
+class UsersController < ApplicationController
+  def new
+      @user = User.new
+      if logged_in?
+        redirect_to tasks_path
+      end
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      session[:user_id] = @user.id
+      redirect_to user_path(@user.id)
+    else
+      render :new
+    end
+  end
+
+  def show
+    @user = User.find(params[:id])
+    unless @user == current_user
+      redirect_to tasks_path
+    end
```

Comment:
この部分も後置if構文の使用が可能です。
```suggestion
    redirect_to tasks_path unless @user == current_user
```